packages/server/src/time-control.ts

```diff
@@ -0,0 +1,90 @@
+import { ObjectId } from "mongodb";
+import { TimeControlType, ITimeControlBase, ITimeControlConfig, IConfigWithTimeControl } from "@ogfcommunity/variants-shared/src/time_control/time_control.types";
+import { gamesCollection } from "./games";
+import { GameResponse } from "@ogfcommunity/variants-shared";
+
+export function HasTimeControlConfig(game_config: unknown): game_config is IConfigWithTimeControl {
+    return (game_config && typeof game_config == "object" && 'time_control' in game_config);
+}
+
+export function ValidateTimeControlConfig(time_control_config: unknown): time_control_config is ITimeControlConfig {
+    return (time_control_config && typeof time_control_config === "object" && 'type' in time_control_config);
+}
+
+export function ValidateTimeControlBase(time_control: unknown): time_control is ITimeControlBase {
+    return (time_control && typeof time_control === "object" && 
+    'moveTimestamps' in time_control && 'remainingMilliseconds' in time_control && 'onThePlaySince' in time_control);
+}
+
+// validation of the config should happen before this is called
+export interface ITimeHandler {
+    handleMove(game: GameResponse, playerNr: number, move: string): Promise<void>;
```

Comment:
I'll be interested to see the variant where different moves affect the time control differently 😄

Reply:
Right, this is unnecessary at the moment. But I think that it does not hurt to include this - who knows what variants we'll come up with in the future :D